apps/events/models.py

```diff
@@ -82,6 +82,30 @@ def get_queryset(self):
             .order_by("-is_today", "registration_filtered")
         )
 
+    def get_queryset_for_user(self, user: User):
+        """
+        :return: Queryset filtered by these requirements:
+            event is visible AND (event has NO group restriction OR user having access to restricted event)
+            OR the user is attending the event themselves
+        """
+        group_restriction_query = Q(group_restriction__isnull=True) | Q(
+            group_restriction__groups__in=user.groups.all()
+        )
+        is_attending_query = (
+            (
+                Q(attendance_event__isnull=False)
+                & Q(attendance_event__attendees__user=user)
+            )
+            if not user.is_anonymous
+            else Q()
+        )
+        is_visible_query = Q(visible=True)
+        return (
+            self.get_queryset()
+            .filter(group_restriction_query & is_visible_query | is_attending_query)
+            .distinct()
+        )
+
```

Comment:
This was previously part of the main event queryset, it was moved to the manager since I thought it fit better there. That code has not been changed